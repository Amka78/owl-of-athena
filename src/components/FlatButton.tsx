import React, { FunctionComponent } from "react";
import { Button as PaperButton } from "react-native-paper";
import {
    MessageLocalizationParam,
    Message,
    Colors,
    Dimens,
    Fonts
} from "../constants";
import { TextStyle, ViewStyle } from "react-native";

type FlatButtonProps = {
    children: MessageLocalizationParam;
    multiLingual?: boolean;
    labelStyle?: TextStyle;
    onPress?: () => void;
};

const labelStyle: TextStyle = {
    color: Colors.white,
    fontFamily: Fonts.primarySemiBold,
    fontSize: Dimens.button_flat_text_size
};
const contentStyle: ViewStyle = {
    marginBottom: Dimens.button_flat_margin_bottom,
    height: Dimens.button_flat_height
};
export const FlatButton: FunctionComponent<FlatButtonProps> = (
    props: FlatButtonProps
) => {
    return (
        <PaperButton
            onPress={props.onPress}
            mode={"text"}
            labelStyle={[labelStyle, props.labelStyle]}
            contentStyle={contentStyle}
        >
            {Message.get(props.children, props.multiLingual)}
        </PaperButton>
    );
};
