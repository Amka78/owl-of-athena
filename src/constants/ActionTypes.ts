const LOGIN = "LOGIN" as const;
const UPDATE_USER = "UPDATE_USER" as const;
const LOGOUT = "LOGOUT" as const;
const CACHE_SETTINGS = "CACHE_SETTINGS" as const;
const INIT = "INIT" as const;
const CACHE_PROFILES = "CACHE_PROFILES" as const;
const CACHE_SESSIONS = "CACHE_SESSIONS" as const;
const SELECT_SESSION = "SELECT_SESSION" as const;
const SELECT_SESSION_DETAIL = "SELECT_SESSION_DETAIL" as const;
const INITIALIZE_AURORA = "INITIALIZE_AURORA" as const;
const INITIALIZE_SESSION = "INITIALIZE_SESSION" as const;
const UPDATE_FILTER = "UPDATE_FILTER" as const;
const DELETE_SESSION = "DELETE_SESSION" as const;
const UPDATE_SESSION = "UPDATE_SESSION" as const;
export default {
    INIT,
    INITIALIZE_AURORA,
    INITIALIZE_SESSION,
    LOGIN,
    LOGOUT,
    UPDATE_USER,
    CACHE_SESSIONS,
    DELETE_SESSION,
    SELECT_SESSION,
    SELECT_SESSION_DETAIL,
    UPDATE_FILTER,
    UPDATE_SESSION,
    CACHE_SETTINGS,
    CACHE_PROFILES,
};
