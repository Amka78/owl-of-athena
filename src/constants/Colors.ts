const teal = "#00ffe6";
const purple = "#4a2fba";
const white = "#ffffff";
export default {
    black: "#000000",
    charcoal: "#333333",
    gray: "#9B9B9B",
    white: white,
    purple: purple,
    navy: "#070724",
    navy_darker: "#09092B",
    blue: "#001438",
    cyan: "#bcfeff",
    teal: teal,
    yellow: "#daf392",
    red: "#ff4310",
    light_orange: "#ff8260",
    semi_transparent: "#66000000",
    aurora_connected: teal,
    aurora_connected_text: purple,
    aurora_disconnected: purple,
    aurora_disconnected_text: white
};
