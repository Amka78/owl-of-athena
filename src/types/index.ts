//#region Import Modules
import { GuestUser } from "./Auth";
import { CreateUser, Login } from "./Login";
import { Signup } from "./Signup";
import { AuroraSound } from "./Sound";
import { User } from "./User";
import { ExperimentalNavigator, WakeLockSentinel } from "./WakeLock";
//#endregion

//#region Types
export {
    AuroraSound,
    CreateUser,
    ExperimentalNavigator,
    GuestUser,
    Login,
    Signup,
    User,
    WakeLockSentinel,
};
//#endregion
