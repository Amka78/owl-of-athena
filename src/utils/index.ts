import BaseUrl from "./BaseUrl";
export { BaseUrl };
