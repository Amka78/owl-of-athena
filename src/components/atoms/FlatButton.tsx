//#region Import Modules
import React, { FunctionComponent } from "react";
import { TextStyle, ViewStyle } from "react-native";
import { Button as PaperButton, useTheme } from "react-native-paper";

import { Dimens, Fonts } from "../../constants";
//#endregion

//#region Types
export type FlatButtonProps = {
    children: string;
    labelStyle?: TextStyle;
    contentStyle?: ViewStyle;
    onPress?: () => void;
};
//#endregion

//#region Component
export const FlatButton: FunctionComponent<FlatButtonProps> = (
    props: FlatButtonProps
) => {
    const theme = useTheme();
    return (
        <PaperButton
            onPress={props.onPress}
            mode={"text"}
            labelStyle={[
                labelStyle,
                { color: theme?.colors.text },
                props.labelStyle,
            ]}
            contentStyle={[contentStyle, props.contentStyle]}
        >
            {props.children}
        </PaperButton>
    );
};
//#endregion

//#region Styles
const labelStyle: TextStyle = {
    fontFamily: Fonts.primarySemiBold,
    fontSize: Dimens.button_flat_text_size,
};
const contentStyle: ViewStyle = {
    marginTop: Dimens.button_margin,
    marginBottom: Dimens.button_flat_margin,
    height: Dimens.button_flat_height,
};
//#endregion
