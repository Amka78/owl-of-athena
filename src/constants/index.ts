import ActionTypes from "./ActionTypes";
import Colors from "./Colors";
import Dimens from "./Dimens";
import Fonts from "./Fonts";
import Layout from "./Layout";
import Message, { MessageLocalizationParam } from "./Message";
import { MessageKeys } from "./MessageKeys";
import StorageKeys from "./StorageKeys";

export {
    ActionTypes,
    Colors,
    Dimens,
    Fonts,
    Layout,
    MessageLocalizationParam,
    Message,
    MessageKeys,
    StorageKeys
};
