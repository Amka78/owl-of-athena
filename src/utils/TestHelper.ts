import "../components";

import { shallow, ShallowWrapper } from "enzyme";
import enzymetoJson from "enzyme-to-json";
import {
    GestureResponderEvent,
    NativeSyntheticEvent,
    TextInputEndEditingEventData,
} from "react-native";

import { ButtonProps } from "../components/atoms/Button";

// @ts-ignore
export const FakeGestureResponderEvent: GestureResponderEvent = {};
// @ts-ignore
export const FakeTextBoxOnEndEditingEventData: NativeSyntheticEvent<TextInputEndEditingEventData> = {};

export function createMock<P, S>(comp: any): ShallowWrapper<P, S> {
    return shallow<P, S>(comp);
}

export function toJson(wrapper: any) {
    return enzymetoJson(wrapper);
}

/**
 * TestIDを利用してComponetのPropsを取得します。
 *
 * @static
 * @template T Propsの型を指定
 * @param {ShallowWrapper} component
 * @param {string} name
 * @param {string} testId
 * @returns {T}
 * @memberof TestHelper
 */
export function getComponentPropsByTestID<T>(
    component: ShallowWrapper<any, any>,
    name: string,
    testId: string
): T | undefined {
    const findComponent = component.find(name).findWhere((wrapper: any) => {
        const targetProps = wrapper.props();
        return targetProps.testID === testId;
    });
    if (findComponent.length <= 0) {
        return undefined;
    }
    const props: T = findComponent.props();
    return props;
}

export function getButtonPropsByTestID(
    component: ShallowWrapper<any, any>,
    testID: string
): ButtonProps | undefined {
    return getComponentPropsByTestID<ButtonProps>(
        component,
        "MultiLingualButton",
        testID
    );
}

/*public static getTextBoxPropsByTestID(
        component: ShallowWrapper<any, any>,
        testID: string
    ): ITextBoxProps | undefined {
        return this.getComponentPropsByTestID<ITextBoxProps>(
            component,
            "MultiLingualTextBox",
            testID
        );
    }

    public static getHeaderSearchBarPropsByTestID(
        component: ShallowWrapper<any, any>,
        testID: string
    ): IHeaderSearchBarProps | undefined {
        return this.getComponentPropsByTestID<IHeaderSearchBarProps>(
            component,
            "HeaderSearchBar",
            testID
        );
    }

    public static getListItemPropsByTestID(
        component: ShallowWrapper<any, any>,
        testID: string
    ): IListItemProps | undefined {
        return this.getComponentPropsByTestID<IListItemProps>(
            component,
            "MultiLingualListItem",
            testID
        );
    }*/

/*public static createNavigationMock(
        param: Partial<NavigationScreenProp<any, any>>
    ): NavigationScreenProp<any, any> {
        const navigation = new Object();
        Object.assign(navigation, param);
        return navigation as NavigationScreenProp<any, any>;
    }*/
