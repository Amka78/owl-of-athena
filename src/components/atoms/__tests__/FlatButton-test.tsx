/* eslint-disable @typescript-eslint/ban-ts-comment */
//#region Import Modules
import "react-native";

import { ShallowWrapper } from "enzyme";
import React from "react";

import { createMock, toJson } from "../../../utils/TestHelper";
import { FlatButton, FlatButtonProps } from "../FlatButton";
import { useTheme } from "react-native-paper";
import { Colors } from "../../../constants";
import { Theme } from "react-native-paper/lib/typescript/types";
//#endregion

//#region Test
let component: ShallowWrapper<FlatButtonProps, unknown, unknown>;
jest.mock("react-native-paper");
const useThemeMock = useTheme as jest.Mock<Theme>;
describe("FlatButton UnitTest", () => {
    afterEach(() => {
        jest.resetAllMocks();
    });
    it("Renders correctly", () => {
        component = createMock(<FlatButton>{"test"}</FlatButton>);

        expect(toJson(component)).toMatchSnapshot();
    });
    it("Overwrite styles", () => {
        // @ts-ignore
        useThemeMock.mockReturnValue({ colors: { text: Colors.red } });
        component = createMock(
            <FlatButton
                contentStyle={{ marginTop: 0 }}
                labelStyle={{ marginTop: 0 }}
            >
                {"test"}
            </FlatButton>
        );

        const props = component.props();
        const labelStyle = props.labelStyle as any;
        const contentStyle = props.contentStyle as any;

        expect(labelStyle[2].marginTop).toBe(0);
        expect(labelStyle[1].color).toBe(Colors.red);
        expect(contentStyle[1].marginTop).toBe(0);
    });
});
//#endregion
