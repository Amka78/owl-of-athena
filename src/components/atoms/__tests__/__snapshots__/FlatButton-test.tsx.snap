// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlatButton UnitTest Renders correctly 1`] = `
<withTheme(Button)
  contentStyle={
    Array [
      Object {
        "height": 28,
        "marginBottom": 12,
        "marginTop": 16,
      },
      undefined,
    ]
  }
  labelStyle={
    Array [
      Object {
        "fontFamily": "calibre_app_semibold",
        "fontSize": 14,
      },
      Object {
        "color": undefined,
      },
      undefined,
    ]
  }
  mode="text"
>
  test
</withTheme(Button)>
`;
